math/src/main/java/com/powsybl/math/graph/UndirectedGraph.java

```diff
@@ -14,62 +14,229 @@
+     * @return the index of the second vertex that the specified edge is connected to.
+     */
     int getEdgeVertex2(int e);
 
+    /**
+     * Remove all the vertices of this graph.
+     * This method throws a {@link com.powsybl.commons.PowsyblException} if edges exist.
+     */
     void removeAllVertices();
 
+    /**
+     * Return an {@link Iterable} to iterate over the attached values to the edges.
+     *
+     * @return an {@link Iterable} to iterate over the attached values to the edges.
+     */
     Iterable<E> getEdgesObject();
 
+    /**
+     * Return a {@link Stream} to iterate over the attached values to the edges.
+     *
+     * @return a {@link Stream} to iterate over the attached values to the edges.
+     */
     Stream<E> getEdgeObjectStream();
 
+    /**
+     * Return the attached value to the specified edge.
+     *
+     * @param e the index of an edge.
+     * @return the attached value to the specified edge.
+     */
     E getEdgeObject(int e);
 
+    /**
+     * Return a {@link List} containing the attached values to the edges between the vertices v1 and v2.
+     *
+     * @return a {@link List} containing the attached values to the edges between the vertices v1 and v2.
+     */
     List<E> getEdgeObjects(int v1, int v2);
 
+    /**
+     * Traverse the entire graph, starting at the specified vertex v.
+     * This method relies on a {@link Traverser} instance to know if the traverse of the graph should continue or stop.
+     * This method throws a {@link com.powsybl.commons.PowsyblException} if the encountered table size is lesser than the vertex count.
+     *
+     * At the end of the method, the encountered array contains {@literal true} for all the traversed vertices, {@literal false} otherwise.
+     *
+     * @param v the vertex index where the traverse has to start.
+     * @param traverser the {@link Traverser} instance to use to know if the traverse should continue or stop.
+     * @param encountered the list of traversed vertices.
+     */
     void traverse(int v, Traverser traverser, boolean[] encountered);
 
+    /**
+     * Traverse the entire graph, starting at the specified vertex v.
+     * This method allocates a boolean array and calls {@link #traverse(int, Traverser, boolean[])}.
+     *
+     * @param v the vertex index where the traverse has to start.
+     * @param traverser the {@link Traverser} instance to use to know if the traverse should continue or stop.
+     */
     void traverse(int v, Traverser traverser);
 
+    /**
+     * Find all paths from the specified vertex.
+     * This method relies on two functions to stop the traverse when the target vertex is found or when an edge must not be traversed.
+     *
+     * @param from the vertex index where the traverse has to start.
+     * @param pathComplete a function that returns true when the target vertex is found.
+     * @param pathCanceled a function that returns true when the edge must not be traversed.
```

Comment:
Not part of the PR, but there is a typo in "cancelled".

Reply:
Fixed